Re-layout of Players Overlay

mousewax opened the pull request:
Added  instructions and inactive player separator to room.

Added CSS to make layout more useful and place your name at the top and active players above inactive players.

ArnoldSmith86 commented:
PR-SERVER-BOT: You can play around with it here: http://212.47.248.129:3761/pr-test (or any other room on that server)

bjalder26 commented:
I think it looks nice, and provides a nice new feature (separating out users playerName) .  However, I'm reluctant to be the one to approve this since I've never worked on UI stuff before and am horrible at CSS.
![image](https://user-images.githubusercontent.com/78324099/134438101-7e8f6b22-0bd2-4ee7-85d2-74f92d1b091c.png)

96LawDawg reviewed (approved):
Well, I will approve it.  I know the least about coding, but ... (1) This is a big improvement in appearance. (2) Wish I had this before because it would have fixed a few issues in games I was playing with inexperienced VTT users.  (3) I want to encourage @mousewax to keep making improvements.

I looked at the CSS and although I don't understand it, I didn't see anything that jumped out at me as potentially a problem.  It also passed all the TestCafe checks, so should be safe to move to production.  However, I think someone else should do the merge just so there is a double check.

mousewax commented:
In the interest of making sure everyone's comfortable, here's a run down of what I did:

The biggest change I made was moving the layout to flexbox rather than just a list of DIVs. In addition to the instructions under the h1, I added an extra separation node to the html.

This allowed me to apply a flexbox "order" to the various classes: myPlayerEntry (order: 1;), then regular divs (order: 2;) that are shortened and get stacked left to right, then the separation node (order: 3;) that's full width to break the active from inactive groups and finally the inactivePlayerEntry (order: 4;) which are likewise shortened to stack left to right.

The "me:" and "other players" titles were added as ::before and ::after pseudo-elements on the myPlayerEntry to keep the markup as simple as possible. 

Lastly, I added some styling to the color input to add the icon and remove the webkit styling.

Because I could order these in CSS, I didn't have to make any changes to the JS that builds the page or change any actual functionality.

RaphaelAlvez commented:
hey I saw that you used WebKit.
This normally creates problems with firefox so I checked and it looks pretty bad.

It doesn't need to be identical or as good and in chrome/safari but maybe in this case a fix is needed. Maybe make the square smaller?

![image](https://user-images.githubusercontent.com/73476656/134538081-ed3133d2-6ef1-4f09-bdb6-c662cb4d6c63.png)

mousewax commented:
I will make sure it is the same in Firefox! Thanks for checking it.

On Thu, Sep 23, 2021 at 8:38 AM RaphaelAlvez ***@***.***>
wrote:

> hey I saw that you used WebKit.
> This normally creates problems with firefox so I checked and it looks
> pretty bad.
>
> It doesn't need to be identical or as good and in chrome/safari but maybe
> in this case a fix is needed. Maybe make the square smaller?
>
> [image: image]
> <https://user-images.githubusercontent.com/73476656/134538081-ed3133d2-6ef1-4f09-bdb6-c662cb4d6c63.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ArnoldSmith86/virtualtabletop/pull/761#issuecomment-925928952>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADB2NRK3TXOHKGYZD3KO66LUDNCVXANCNFSM5ESOV7LQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

RaphaelAlvez commented on client/css/overlays/players.css:
```diff
@@ -1,23 +1,68 @@
+  height: 2rem;
+}
+
+.teamColor::-webkit-color-swatch-wrapper {
+  padding: 0;
+}
+
+.teamColor::-webkit-color-swatch {
+  border-radius: 100%;
+  border: 1px solid rgba(255, 255, 255, 0.25);
+  /* background properties broken up to avoid erasing the selected colors */
+  background-image: url(/i/icons-white/user-solid-svg.svg);
+  background-repeat: no-repeat;
+  background-position: center center;
+  background-size: 1rem;
 }
```
Webkit does not work on firefox

mousewax replied:
The appropriate -moz equivalents were added, which I didn't knew were things. Usually it's a struggle getting around all of Apple's ornamentation. I look forward to a day where we get rid of vendor prefixes.

RaphaelAlvez reviewed (changes requested):
Just take a look at this. Is there anything to do about it?

looks great on safari and chrome. in general its a good improvement

mousewax commented:
Gecko does not add the extraneous styling that webkit’s does and *generally
*doesn’t need those extra bits so there’s nothing to do for Firefox there.
But I definitely need to override other things on Firefox.

On Thu, Sep 23, 2021 at 8:42 AM RaphaelAlvez ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Just take a look at this. Is there anything to do about it?
>
> looks great on safari and chrome. in general its a good improvement
> ------------------------------
>
> In client/css/overlays/players.css
> <https://github.com/ArnoldSmith86/virtualtabletop/pull/761#discussion_r714925110>
> :
>
> > +.teamColor::-webkit-color-swatch-wrapper {
> +  padding: 0;
> +}
> +
> +.teamColor::-webkit-color-swatch {
> +  border-radius: 100%;
> +  border: 1px solid rgba(255, 255, 255, 0.25);
> +  /* background properties broken up to avoid erasing the selected colors */
> +  background-image: url(/i/icons-white/user-solid-svg.svg);
> +  background-repeat: no-repeat;
> +  background-position: center center;
> +  background-size: 1rem;
>  }
>
> Webkit does not work on firefox
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ArnoldSmith86/virtualtabletop/pull/761#pullrequestreview-762196715>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADB2NRPRVPB2EDTMF3NCG6DUDNDHRANCNFSM5ESOV7LQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

mousewax commented:
Gecko now adds extraneous styling and pseudo elements. I just committed the styles to cover for it. Gecko and WebKit now look identical.

![image](https://user-images.githubusercontent.com/12822213/134599673-c7b086d4-955c-4f57-ae48-8b2d831564fd.png)

bjalder26 commented:
They look virtually identical to me (and pretty amazing).
Click update branch -> then I think RaphaelAlvez needs to resolve the change request -> then wait 24 hrs before merging.
